OPHD/States/MapViewStateHelper.cpp

```diff
@@ -578,14 +578,14 @@ NAS2D::Xml::XmlElement* writeResearch(const ResearchTracker& tracker)
 
 	research->attribute("completed_techs", completedResearch);
 
-	for (auto& item : tracker.currentResearch())
+	for (const auto& [techId, values] : tracker.currentResearch())
 	{
 		research->linkEndChild(NAS2D::dictionaryToAttributes(
 			"current",
 			{{
-				{"tech_id", item.first},
-				{"progress", std::get<0>(item.second)},
-				{"assigned", std::get<1>(item.second)},
+				{"tech_id", techId},
+				{"progress", std::get<0>(values)},
+				{"assigned", std::get<1>(values)},
```

Comment:
I wonder if you can nest structured bindings to get rid of the `std::get<0>` and `std::get<1>`. Or, just use a `struct` rather than a `std::tuple`.

Reply:
I tried, it yelled at me. I can go back to a struct, I just didn't think I'd ever use more than the two and tend to prefer using already available structures/classes versus providing my own.

Granted I agree `get<0>` and `get<1>` is _way_ less intuitive than named fields in a custom `struct`... so perhaps it's worth it.